### PR TITLE
CHI-1578: Queue Transfer

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

* Updates behaviour of chat transfers to treat transfers to queues differently to transfers to counsellors
* Move .prettierrc settings into package.jsons because a bug in webstorm means it doesn't pick them up from .prettierrc

The updated logic means that a 'queue' target transfer will complete the original task as soon as it hits a queue, rather than holding it open until accepted by a worker, which worker transfers do

It also limits the special logic for rejections to worker transfers. Queue transfers just go back on the queue if rejected now

### Checklist
- [X] Corresponding issue has been opened
- [x] New tests added

### Related Issues
Fixes CHI-1578

### Verification steps

See ticket

### CODE REVIEW

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 1]
```diff
@@ -29,6 +31,11 @@ type TransferMeta = {
   transferStatus: 'transferring' | 'accepted' | 'rejected';
 };
 
+type ChatTransferTaskAttributes = {
+  transferMeta?: TransferMeta;
+  transferTargetType: 'worker' | 'queue';
+};
+
 const isChatTransfer = (
```
Comment (GPaoloni) [inline functions/taskrouterListeners/transfersListener.private.ts:39]: Should `taskAttributes.transferMeta.transferStatus === 'accepted'` be removed from this function? Or renamed the function?

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 2]
```diff
@@ -41,16 +48,29 @@ const isChatTransfer = (
...
 
 const isChatTransferRejected = (
   eventType: EventType,
   taskChannelUniqueName: string,
-  taskAttributes: { transferMeta?: TransferMeta },
+  taskAttributes: ChatTransferTaskAttributes,
 ) =>
   (eventType === RESERVATION_REJECTED || eventType === RESERVATION_TIMEOUT) &&
-  isChatTransfer(taskChannelUniqueName, taskAttributes);
+  isChatTransfer(taskChannelUniqueName, taskAttributes) &&
```
Comment (GPaoloni) [inline functions/taskrouterListeners/transfersListener.private.ts:63]: Won't this assume that `taskAttributes.transferMeta.transferStatus === 'accepted'`?

**Review by GPaoloni — APPROVED** [review]
Changes lgtm. Just small questions about when we consider the transfer "accepted" or "rejected".

File: functions/taskrouterListeners/transfersListener.private.ts (hunk 2 above)
Comment (stephenhand) [reply functions/taskrouterListeners/transfersListener.private.ts:63]: So apparently in flex, we start all our cold transfers with a status of 'accepted', which is causing me some headaches for determining whether to show the transfer buttons or not... 

We could start using the 'transferring' state for cold transfers, but then we would have to remove the check here. Could that have unexpected ramifications?

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 3]
```diff
@@ -41,16 +48,29 @@ const isChatTransfer = (
...
+  taskAttributes.transferTargetType === 'worker';
+
+const isChatTransferToQueueComplete = (
+  eventType: EventType,
+  taskChannelUniqueName: string,
+  taskAttributes: { transferMeta?: TransferMeta; transferTargetType: 'worker' | 'queue' },
```
Comment (murilovmachado) [inline functions/taskrouterListeners/transfersListener.private.ts:69]: Here you can use the type you created: `ChatTransferTaskAttributes`

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 4]
```diff
@@ -41,16 +48,29 @@ const isChatTransfer = (
 const isChatTransferAccepted = (
   eventType: EventType,
   taskChannelUniqueName: string,
-  taskAttributes: { transferMeta?: TransferMeta },
-) => eventType === RESERVATION_ACCEPTED && isChatTransfer(taskChannelUniqueName, taskAttributes);
+  taskAttributes: ChatTransferTaskAttributes,
+) =>
+  eventType === RESERVATION_ACCEPTED &&
+  isChatTransfer(taskChannelUniqueName, taskAttributes) &&
+  taskAttributes.transferTargetType === 'worker';
```
Comment (murilovmachado) [inline functions/taskrouterListeners/transfersListener.private.ts:55]: Maybe renaming these functions to `isChatTransferToWorkerAccepted` and `isChatTransferToWorkerRejected`?

**Review by murilovmachado — APPROVED** [review]
Great to see the changes were "simple". I was worried it could be messier, but I was wrong. Also, checked the behavior and it seems fine.

**Pull request merged**